### ISSUE TITLE
Issue with metaclass and Schemas in Nested fields

### ISSUE THREAD

**Issue by sbrieuc** [apryor6/flask_accepts#66]
Hi,

This refers to #39 and #40 

I'm in a situation where I use flask_accepts in association with flask_marshmallow and marshmallow_sqlalchemy.

Basically, all schemas that I define are not constructed from marshmallow base Schema class directly but from another class SQLAlchemySchema (which itself inherits from marshmallow base Schema). In that configuration, when trying to use these Schemas in a Nested field, flask_accepts throws a TypeError as seen in #39: _"Unknown type for marshmallow model field was used."_

Long story short after looking into it, I can see that SQLAlchemySchema class uses its own metaclass (inherited from marshmallow SchemaMeta) which prevents #40 from working as it uses `isinstance` to check the type of the provided Schema.

In example (simplified), we have:
 ```python
from marshmallow.schema import Schema, SchemaMeta


class SQLAlchemyMeta(SchemaMeta):
    ...

class SQLAlchemySchema(Schema, metaclass=SQLAlchemyMeta):
    ...
```
which leads to the following:
```python
>>> isinstance(type(SQLAlchemySchema), SchemaMeta)
False
```
and causes the error because `type(SQLAlchemySchema)` returns `SQLAlchemyMeta` and not `SchemaMeta`.

I think we can solve this by using `issubclass` instead of `isinstance` in #40, which would give:
```python
def map_type(val, api, model_name, operation):
    value_type = type(val)
    if value_type in type_map:
        return type_map[value_type](val, api, model_name, operation)

    if issubclass(value_type, SchemaMeta) or issubclass(value_type, Schema):
        return type_map[Schema](val, api, model_name, operation)

    raise TypeError('Unknown type for marshmallow model field was used.')
```

I've implemented this change in my project and all seems to work fine. 

I'm more than willing to make a PR for this but I'd like to have feedback on my analysis and make sure I did not miss anything.

**Comment by sbrieuc** [comment]
I'm noticing that the issue occurs only when using the Schema class and not with an instance.

Does not work:
```python
class Foo(Schema):
    pass

class Bar(SQLAlchemySchema):
    items = fields.Nested(Foo)
```

Works:
```python
class Foo(Schema):
    pass

class Bar(SQLAlchemySchema):
    items = fields.Nested(Foo())
```

So I'm not sure of the direction to take here: always providing an instance of the schema will do the trick and we can leave it as it is, but I feel that it's not necessarily the best practice in term of memory usage. What do you think ?

**Comment by apryor6** [comment]
I totally agree that `issubclass` would be more suitable and would be more than happy to accept a PR. Just add a test that asserts behavior works as intended if you provide either type of Schema and we should be good to go.
